crates/proto/src/udp/udp_stream.rs

```diff
@@ -213,14 +217,20 @@ pub(crate) struct NextRandomUdpSocket<P: RuntimeProvider> {
     attempted: usize,
     #[allow(clippy::type_complexity)]
     future: Option<Pin<Box<dyn Send + Future<Output = io::Result<P::Udp>>>>>,
+    avoid_local_ports: Option<Arc<HashSet<u16>>>,
```

Comment:
I think we should skip the `Option` wrapper at this layer, and probably in all non-public API? Could even argue that this is straightforward enough (probably just `Arc::default()`) that we don't even need an `Option` wrapper at all?